app/create_app.py

```diff
@@ -47,6 +114,8 @@ def mark_complete(item_id):
 
 
     @app.route('/delete/<item_id>', methods=['GET'])
+    @login_required
+    @authorized_for('writer')
     def delete_item(item_id):
```

Comment:
Having GET requests modify state can be insecure, e.g. what if I get you to click on a link like <https://devops-starter.herokuapp.com/delete/some-item-id> when you're signed in to the app? Modern browsers don't send cookies with cross-site POST requests from forms, but will with GET links.